### PR TITLE
[Python] Add `__getattr__` and `__getitem__` implementations for DuckDBPyRelation

### PR DESCRIPTION
**Pull request by Tishj**
This adds the following syntax:
`relation.column_of_relation`
`relation['column_of_relation']`

Example:
```py
import duckdb

duckdb.sql('create table tbl (name varchar, age int, date_of_birth timestamp)')
rel = duckdb.table('tbl')
# Create a projection on the table only selecting the name column
name_relation = rel.name
# Does the same as above
also_name_relation = rel['name']
```

### CODE REVIEW

**Review by Mytherin — APPROVED** [review]
Thanks for the PR! LGTM

**Comment by Tishj** [comment]
Not sure what's up with this failure in <https://github.com/duckdb/duckdb/actions/runs/4365008302/jobs/7635131271>

It fails on `OSError: zlib inflate failed: incorrect header check` in the H2OAI tests, I would guess it's a network issue and the file wasn't downloaded properly?

**Comment by Mytherin** [comment]
Yep that's a network issue

File: tools/pythonpkg/src/include/duckdb_python/pyrelation.hpp [hunk 1]
```diff
@@ -52,6 +52,8 @@ struct DuckDBPyRelation {
 
 	void Close();
 
+	unique_ptr<DuckDBPyRelation> GetAttribute(const string &name);
```
Comment (Mause) [inline tools/pythonpkg/src/include/duckdb_python/pyrelation.hpp:55]: I assume this just throws a `TypeError` if you do `relation[None]`?

**Review by Mause — APPROVED** [review]

File: tools/pythonpkg/src/include/duckdb_python/pyrelation.hpp (hunk 1 above)
Comment (Tishj) [reply tools/pythonpkg/src/include/duckdb_python/pyrelation.hpp:55]: Yea it can't implicitly convert it to string, but might be worth a test 👍

**Pull request merged**